base/REPL.jl

```diff
@@ -831,7 +831,27 @@ function setup_interface(repl::LineEditREPL; hascolor = repl.hascolor, extra_rep
             input = takebuf_string(sbuffer)
             oldpos = start(input)
             firstline = true
+            isprompt_paste = false
             while !done(input, oldpos) # loop until all lines have been executed
+                # Check if the next statement starts with "julia> ", in that case
+                # skip it. But first skip whitespace
+                while input[oldpos] in ('\n', ' ', '\t')
+                    oldpos = nextind(input, oldpos)
+                    oldpos >= sizeof(input) && return
+                end
+                # Check if input line starts with "julia> ", remove it if we are in prompt paste mode
+                jl_prompt_len = 7
+                 if (firstline || isprompt_paste) && (oldpos + jl_prompt_len <= sizeof(input) && input[oldpos:oldpos+jl_prompt_len-1] == "julia> ")
```

Comment:
Perhaps. What is possible to do is to remove an initial prompt if it exists in a pasted text but multiple statements and scrubbing away output is only possible because the julia parser tells us when an expression ends.